cmd/sovereignnode/chainSimulator/chainSimulator.go

```diff
@@ -0,0 +1,99 @@
+package chainSimulator
+
+import (
+	"path"
+
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/config"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	"github.com/multiversx/mx-chain-go/factory"
+	"github.com/multiversx/mx-chain-go/factory/runType"
+	chainSimulatorIntegrationTests "github.com/multiversx/mx-chain-go/integrationTests/chainSimulator"
+	"github.com/multiversx/mx-chain-go/node"
+	"github.com/multiversx/mx-chain-go/node/chainSimulator"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/process/rating"
+	"github.com/multiversx/mx-chain-go/sharding"
+	sovereignConfig "github.com/multiversx/mx-chain-go/sovereignnode/config"
+	"github.com/multiversx/mx-chain-go/sovereignnode/incomingHeader"
+	sovRunType "github.com/multiversx/mx-chain-go/sovereignnode/runType"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+)
+
+// ArgsSovereignChainSimulator holds the arguments for sovereign chain simulator
+type ArgsSovereignChainSimulator struct {
+	SovereignExtraConfig config.SovereignConfig
+	ChainSimulatorArgs   chainSimulator.ArgsChainSimulator
```

Comment:
Should've made this a pointer inheritance.
```suggestion
	  *chainSimulator.ArgsChainSimulator
```